### PR TITLE
usb audio changes

### PR DESCRIPTION
**Pull request by jzipperer-fb**
Added device tree bindings for some addition audio usb descriptor values
Added start of frame notifications to nxp mcux usb

### CODE REVIEW

File: subsys/usb/device/class/audio/usb_audio_internal.h [hunk 1]
```diff
@@ -165,14 +165,26 @@
...
+
 #define SYNC_TYPE_HP(i)     3
 #define SYNC_TYPE_MIC(i)    DT_ENUM_IDX(DT_INST(i, COMPAT_MIC), sync_type)
 #define SYNC_TYPE_HS_HP(i)  3
 #define SYNC_TYPE_HS_MIC(i) DT_ENUM_IDX(DT_INST(i, COMPAT_HS), mic_sync_type)
 #define SYNC_TYPE(dev, i) (SYNC_TYPE_##dev(i) << 2)
 
 #define EP_SIZE(dev, i) \
-	((GET_RES(dev, i)/8) * CH_CNT(dev, i) * 48)
+	((GET_RES(dev, i)/8) * CH_CNT(dev, i) * (GET_RATE(dev, i)/1000))
```
Comment (tmon-nordic) [inline subsys/usb/device/class/audio/usb_audio_internal.h:187]: This doesn't account for sample rates that are not full multiple of 1000, e.g. 44100. However to support 44100 properly also changes in other parts of the class implementation would be needed.

Considering that I am actively working on UAC2 implementation for the new stack, I am not sure if it is worthwhile to try to fix the old USB Audio implementation.

Comment (jzipperer-fb) [reply subsys/usb/device/class/audio/usb_audio_internal.h:187]: On my side I'm actually using 3.4.0, would it be worth targeting this as a change to 3.4.0 instead?  What is the ETA for UAC2 support?

Comment (tmon-nordic) [reply subsys/usb/device/class/audio/usb_audio_internal.h]: I hope is that I will manage to open the PR with UAC2 + explicit feedback sample in January. The implicit feedback sample will follow later (the remaining complexity is on the sample application side, the class is pretty much ready for implicit feedback). I will CC you when I open the PR.

**Review by tmon-nordic — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

File: drivers/usb/device/usb_dc_mcux.c [hunk 2]
```diff
@@ -910,6 +910,9 @@ static void usb_mcux_thread_main(void *arg1, void *arg2, void *arg3)
 		case kUSB_DeviceNotifyResume:
 			dev_state.status_cb(USB_DC_RESUME, NULL);
 			break;
+		case kUSB_DeviceNotifySOF:
```
Comment (mmahadevan108) [inline drivers/usb/device/usb_dc_mcux.c:913]: Where is `kUSB_DeviceNotifySOF`. This is causing CI failures.

Comment (jzipperer-fb) [reply drivers/usb/device/usb_dc_mcux.c:913]: Oh no!  It's in the nxp middleware usb sdk.  device/usb_device_dci.h.

Comment (jzipperer-fb) [reply drivers/usb/device/usb_dc_mcux.c:913]: https://github.com/nxp-mcuxpresso/mcux-sdk-middleware-usb/pull/5

Comment (mmahadevan108) [reply drivers/usb/device/usb_dc_mcux.c:913]: @jzipperer-fb, the NXP SDK team has to review and integrate these changes first. We can then add these changes to Zephyr. For now, I have added a PR to revert the 2 commits to the NXP drivers.

Comment (jzipperer-fb) [reply drivers/usb/device/usb_dc_mcux.c:913]: ok.  sounds good.

Comment (jzipperer-fb) [reply drivers/usb/device/usb_dc_mcux.c:913]: how can i kick start that process?

Comment (mmahadevan108) [reply drivers/usb/device/usb_dc_mcux.c:913]: I have followed up with the SDK team via the PR you submitted. We can keep an eye on that PR.

Comment (jsiverskog) [reply drivers/usb/device/usb_dc_mcux.c:913]: hey,
this was merged to main, causing build issues:
```
drivers/usb/device/usb_dc_mcux.c: In function 'usb_mcux_thread_main':
drivers/usb/device/usb_dc_mcux.c:913:22: error: 'kUSB_DeviceNotifySOF' undeclared (first use in this function); did you mean 'kUSB_DeviceNotifyError'?
  913 |                 case kUSB_DeviceNotifySOF:
      |                      ^~~~~~~~~~~~~~~~~~~~
      |                      kUSB_DeviceNotifyError
```
i believe these changes should be reverted, https://github.com/nxp-mcuxpresso/mcux-sdk-middleware-usb/pull/5 patch be merged into https://github.com/zephyrproject-rtos/hal_nxp/ and finally the manifest updated in this repo?

Comment (jsiverskog) [reply drivers/usb/device/usb_dc_mcux.c:913]: ah, saw https://github.com/zephyrproject-rtos/zephyr/pull/67459 now.

**Pull request merged**